providers/windows/host_windows.go

```diff
@@ -137,6 +143,49 @@ func (r *reader) hostname(h *host) {
 	h.info.Hostname = v
 }
 
+func (r *reader) fqdn(h *host) {
+	fqdn, err := getComputerNameEx(
+		stdwindows.ComputerNamePhysicalDnsFullyQualified)
+	if err != nil {
+		r.addErr(fmt.Errorf("could not get windows FQDN: %s", err))
+		return
+	}
+
+	h.info.FQDN = strings.TrimSuffix(fqdn, ".")
+}
+
+func getComputerNameEx(name uint32) (string, error) {
+	size := uint32(64)
+
+	for {
+		buff := make([]uint16, size)
+		err := stdwindows.GetComputerNameEx(
+			name, &buff[0], &size)
+		if err == nil {
+			return syscall.UTF16ToString(buff[:size]), nil
+		}
+
+		// ERROR_MORE_DATA means buff is too small and size is set to the
+		// number of bytes needed to store the FQDN. For details, see
+		// https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getcomputernameexw#return-value
+		if errors.Is(err, syscall.ERROR_MORE_DATA) {
+			if size <= uint32(len(buff)) {
+				// Safeguard to avoid an infinite loop.
+				return "", fmt.Errorf(
+					"windows.GetComputerNameEx returned ERROR_MORE_DATA, " +
+						"but data size should fit into buffer")
+			} else {
+				// Grow the buffer and try again.
+				// Should we limit its growth?
```

Comment:
yes

Reply:
but then what would be the limit?
`size` is the needed size for a buffer to fit the FQDN. The unix implementation does not limit the size, so I'd say it should be fine. it hardly wouldn't fit into memory, if it don't, I'd say windows is on a lot more trouble than us.